### PR TITLE
Support cross thread trace profiling

### PR DESCRIPTION
**Pull request by mrproliu**
Following https://github.com/apache/skywalking/issues/10373, support the query of the cross thread trace profiling. 

Also, the `readNullableMetricsValue` query been added with default implementation, this should be update in other PR. 

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Is this going to break UI temporarily, right?

**Comment by mrproliu** [comment]
> Is this going to break UI temporarily, right?

Yes, It need the UI to adapt with the new query protocol.

**Comment by wu-sheng** [comment]
FYI @apache/skywalking-committers After this gets merged, we are going to hold the demo update until we have UI side adoption. Otherwise, the profiling relative features are broken.

File: oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/MetricQuery.java [hunk 1]
```diff
@@ -92,6 +93,14 @@ public IntValues getLinearIntValues(final MetricCondition metrics,
         return metricsValues.getValues();
     }
 
+    public NullableValue readNullableMetricsValue(MetricsCondition condition, Duration duration) {
+        // TODO default implantation
```
Comment (wu-sheng) [inline oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/MetricQuery.java:97]: FYI @wankai123

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskQueryService.java [hunk 2]
```diff
@@ -161,38 +182,126 @@ public List<ProfileTaskLog> getProfileTaskLogs(final String taskID) throws IOExc
...
+        // build result
+        return instanceTraceWithSegments.values().stream()
+            .flatMap(s -> buildProfiledSegmentsList(s, profiledSegmentIdList).stream())
+            .collect(Collectors.toList());
+    }
 
-        ProfiledSegment profiledSegment = new ProfiledSegment();
-        SegmentObject segmentObject = SegmentObject.parseFrom(segmentRecord.getDataBinary());
-        profiledSegment.getSpans().addAll(buildProfiledSpanList(segmentObject));
+    protected List<ProfiledTraceSegments> buildProfiledSegmentsList(List<SegmentRecord> segmentRecords, List<String> profiledSegmentIdList) {
```
[the remark this answers is not in the document]
Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskQueryService.java:238]: fixed.

**Comment by wu-sheng** [comment]
@mrproliu There is a snapshot exporter tool. Do we need some update about that?

**Comment by mrproliu** [comment]
> @mrproliu There is a snapshot exporter tool. Do we need some update about that?

The tools have been updated to support multiple segments. If users want to use them, they need to update them too.

**Comment by wu-sheng** [comment]
Is there anything expected to change for https://skywalking.apache.org/docs/main/next/en/guides/backend-profile-export/?

Notice, the profiling docs should be moved from contribute doc to new profiling doc. https://skywalking.apache.org/docs/main/next/en/guides/readme/#profile
Your next move should be about docs, right?

**Review by wu-sheng — DISMISSED** [review]

**Comment by mrproliu** [comment]
> Is there anything expected to change for https://skywalking.apache.org/docs/main/next/en/guides/backend-profile-export/?

No, this file only descript how to use this export tool, and the usage has no change. 

> Your next move should be about docs, right?

Yes, I will write the documentation for cross-thread profiling and continuous profiling.

File: docs/en/changes/changes.md [hunk 3]
```diff
@@ -17,6 +17,7 @@
 * Support continuous profiling feature.
 * Support collect process level related metrics.
 * Fix K8sRetag reads the wrong k8s service from the cache due to a possible namespace mismatch.
+* Support cross thread trace profiling.
```
Comment (wu-sheng) [inline docs/en/changes/changes.md:20]:
```suggestion
* [Breaking Change] Support cross-thread trace profiling. The data structure and query APIs are changed.
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**